src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1945,6 +1949,73 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
       warning().source_location=f_op.find_source_location();
       warning() << "function `" << identifier << "' is not declared" << eom;
     }
+    else if(
+      sym_entry->second.type.get_bool(ID_C_inlined) &&
+      sym_entry->second.is_macro && sym_entry->second.value.is_not_nil())
+    {
+      // calling a function marked as always_inline
```

Comment:
Or, presumably, a macro? Or is the `is_macro` flag only used for functions that are not in fact macros?

Reply:
Yes, `is_macro` isn't ever used with (preprocessor) macros.